homework/src/index.js

```diff
@@ -30,18 +33,70 @@
     return elem;
   }
 
+  function renderRepository(container, repository) {
+    container.innerHTML = '';
+    //const root = document.getElementById('root');
+    const table = createAndAppend('table', container);
+    const tBody = createAndAppend('tBody', table);
+    const trRepository = createAndAppend('tr', tBody);
+    trRepository.className = "table-info";
+    createAndAppend('td', trRepository, {html: "Repository:"});
+    //createAndAppend('td', trRepository, {html: repository.name});
+    const link = createAndAppend('a', trRepository,{html: repository.name});
+    link.setAttribute('href', repository.html_url);
+    const trDescripton = createAndAppend('tr', tBody);
+    createAndAppend('td', trDescripton, {html: "Description:"});
+    createAndAppend('td', trDescripton,{html: repository.description});
+    trDescripton.className = "table-info";
+    const trForks = createAndAppend('tr', tBody);
+    createAndAppend('td', trForks, {html: "Forks:"});
+    createAndAppend('td', trForks,{html: repository.forks});
+    trForks.className = "table-info";
+     const trUpdated = createAndAppend('tr', tBody);
+    createAndAppend('td', trUpdated, {html: "Updated:"});
+    createAndAppend('td', trUpdated, {html: repository.updated_at.toLocaleString()});
+    trUpdated.className = "table-info";
+   // createAndAppend('pre', container, { html: JSON.stringify(repository, null, 2)});
+  }
+
   function main(url) {
-    fetchJSON(url, (err, data) => {
+   
       const root = document.getElementById('root');
+      root.className = "root";
+      //createAndAppend('pre', root, {html: JSON.stringify(repositories, null, 2)})
+      const div = createAndAppend('div', root);
+      div.className ="headbox";
+      const heading = createAndAppend('heading', div, {html: "HYF Repositories"});
+      heading.className = "heading";
+      const select = createAndAppend('select',div);
+      select.className = "select";
+      const container = createAndAppend('div', root);
+      container.className = "container";
+      // const container1 = createAndAppend('div', root, {html: "Contributions" });
+      // container1.className = "container1";
+      
+      fetchJSON(url, (err, repositories) => {
+      repositories.forEach((repository, index) => {
+        createAndAppend('option', select, {html: repository.name, value: index});
+      });
+      select.addEventListener('change', (event)=> {
+        renderRepository(container, repositories[event.target.value]);
+      });
+      renderRepository(container, repositories[0]);
+    });
+  
+
       if (err) {
```

Comment:
The variable `err` is undefined here. This is the error message I get when I open the Chrome Developer Tools:

```
Uncaught ReferenceError: err is not defined
    at main (index.js:89)
    at window.onload (index.js:99)
```

Furthermore, in VSCode this variable is underlined, warning about it being not defined.

Needless to say that your error handling does not work.

In fact, this whole code block with the if-then-else looks like a left-over from the skeleton code.